starmap/specifications/primitives/gesture_recognizers/GestureRecognizer.md

```diff
@@ -68,40 +68,20 @@ protocol GestureRecognizer {
   var enabled: Bool
 ```
 
-### States
+### State changes
 
-A gesture recognizer can be in any one of the following states:
+A gesture recognizer is notified of the following state change events:
 
 ```
-enum GestureRecognizerState {
-  case Possible
-  case Began
-  case Changed
-  case Ended
-  case Cancelled
-  case Failed
-
-  case Recognized = Ended
+protocol StateChangeObserver {
+  func didBegin(centroid: Point)
+  func didMove(centroid: Point)
```

Comment:
I sort of agree with the conclusion, but not so much with the rationale - the user of the API will just ignore `centroid` if they do not care about it.

The rationale I'm thinking of is that there may be multiple outputs - centroid, velocity, pressure, and it becomes increasingly unmanageable to stick all those in as parameters rather than exposing getters.